frontend/packages/console-shared/src/promql/resource-metrics.ts

```diff
@@ -0,0 +1,152 @@
+  ),
+};
+
+export const getPodUtilizationQueries = (
+  name: string,
+  namespace: string,
+): { [key: string]: string[] } => ({
+  [ResourceMetricsQuery.RESOURCE_MEMORY_UTILIZATION]: [
+    podUtilizationQueries[ResourceMetricsQuery.RESOURCE_MEMORY_UTILIZATION]({ name, namespace }),
+    podUtilizationQueries[ResourceMetricsQuery.RESOURCE_QUOTA_LIMIT]({
+      name,
+      namespace,
+      resource: 'memory',
+    }),
+    podUtilizationQueries[ResourceMetricsQuery.RESOURCE_QUOTA_REQUEST]({
+      name,
+      namespace,
+      resource: 'memmory',
+    }),
+  ],
+  [ResourceMetricsQuery.RESOURCE_CPU_UTILIZATION]: [
+    podUtilizationQueries[ResourceMetricsQuery.RESOURCE_CPU_UTILIZATION]({ name, namespace }),
+    podUtilizationQueries[ResourceMetricsQuery.RESOURCE_QUOTA_LIMIT]({
+      name,
+      namespace,
+      resource: 'cpu',
+    }),
+    podUtilizationQueries[ResourceMetricsQuery.RESOURCE_QUOTA_REQUEST]({
+      name,
+      namespace,
+      resource: 'cpu',
+    }),
+  ],
+  [ResourceMetricsQuery.RESOURCE_FILESYSTEM_UTILIZATION]: [
+    podUtilizationQueries[ResourceMetricsQuery.RESOURCE_FILESYSTEM_UTILIZATION]({
+      name,
+      namespace,
+    }),
+  ],
+  [ResourceMetricsQuery.RESOURCE_NETWORK_UTILIZATION_IN]: [
+    podUtilizationQueries[ResourceMetricsQuery.RESOURCE_NETWORK_UTILIZATION_IN]({
+      name,
+      namespace,
+    }),
+  ],
+  [ResourceMetricsQuery.RESOURCE_NETWORK_UTILIZATION_OUT]: [
+    podUtilizationQueries[ResourceMetricsQuery.RESOURCE_NETWORK_UTILIZATION_OUT]({
+      name,
+      namespace,
+    }),
+  ],
+});
+
+export const getPodControllerUtilizationQueries = (
+  name: string,
+  namespace: string,
+  type: string,
+): { [key: string]: string[] } => ({
+  [ResourceMetricsQuery.RESOURCE_MEMORY_UTILIZATION]: [
+    podControllerUtilizationQueries[ResourceMetricsQuery.RESOURCE_MEMORY_UTILIZATION]({
+      name,
+      namespace,
+      type,
+    }),
+  ],
+  [ResourceMetricsQuery.RESOURCE_CPU_UTILIZATION]: [
+    podControllerUtilizationQueries[ResourceMetricsQuery.RESOURCE_CPU_UTILIZATION]({
+      name,
+      namespace,
+      type,
+    }),
+  ],
+  [ResourceMetricsQuery.RESOURCE_FILESYSTEM_UTILIZATION]: [
+    podControllerUtilizationQueries[ResourceMetricsQuery.RESOURCE_FILESYSTEM_UTILIZATION]({
+      name,
+      namespace,
+      type,
+    }),
+  ],
+  [ResourceMetricsQuery.RESOURCE_NETWORK_UTILIZATION_IN]: [
+    podControllerUtilizationQueries[ResourceMetricsQuery.RESOURCE_NETWORK_UTILIZATION_IN]({
+      name,
+      namespace,
+      type,
+    }),
+  ],
+  [ResourceMetricsQuery.RESOURCE_NETWORK_UTILIZATION_OUT]: [
+    podControllerUtilizationQueries[ResourceMetricsQuery.RESOURCE_NETWORK_UTILIZATION_OUT]({
+      name,
+      namespace,
+      type,
+    }),
+  ],
+});
+
+export const getResourceUtilizationQueries = (obj: K8sResourceKind, model: K8sKind) => {
```

Comment:
nit: We could determine the type from `obj.apiVersion` and `obj.kind` if we wanted to remove one of the function arguments.